lib/core/src/lib/datatable/services/datatable.service.ts

```diff
@@ -0,0 +1,31 @@
+/*!
+ * @license
+ * Copyright 2019 Alfresco Software, Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Injectable } from '@angular/core';
+import { Subject } from 'rxjs';
+import { DataRowUpdateModel } from '../data/data-row-update.model';
+
+@Injectable({
+    providedIn: 'root'
```

Comment:
Actually, I've just realised, this auto provision should not be here, since this way there would be a default singleton instance, which:
- makes the Optional() flag meaningless
- makes it very difficult to troubleshoot in case of multiple datatables